cssom-view-1/Overview.bs

```diff
@@ -1806,6 +1806,8 @@ interface VisualViewport : EventTarget {
 
   readonly attribute double scale;
 
+  readonly attribute FrozenArray&lt;DOMRect> segments;
```

Comment:
I don't think you need the `&lt;`?

Reply:
I don't think a `FrozenArray` can be used tho, right? A `FrozenArray` is supposed to be immutable (https://webidl.spec.whatwg.org/#idl-frozen-array), but this is mutable right?

Reply:
I think it should just be a `sequence`?

Reply:
> I don't think you need the `&lt;`?

I get an error otherwise.

Reply:
> I don't think a `FrozenArray` can be used tho, right? A `FrozenArray` is supposed to be immutable (https://webidl.spec.whatwg.org/#idl-frozen-array), but this is mutable right?

It's frozen because it's not changing, just like any other property of visualviewport. You're expected to listen for some events to query the new state (e.g. resize). If just the posture is changed then [Device Posture API](https://www.w3.org/TR/device-posture/) has an event listener specifically for that. Both APIs are complimentary.

Reply:
> I think it should just be a `sequence`?

Probably a good idea since it can be zero length.

Reply:
> It's frozen because it's not changing, just like any other property of visualviewport. You're expected to listen for some events to query the new state (e.g. resize). If just the posture is changed then [Device Posture API](https://www.w3.org/TR/device-posture/) has an event listener specifically for that. Both APIs are complimentary.

Well but that means that they do change (when the event fires) right? or what am I missing?

Reply:
Here is a concrete example: initial page load and the device is used flat (or a laptop posture with the keyboard docked) the `segments` property will be 0 meaning an array of 0 length. 

If I change the posture to folded (or detach the keyboard) then the events will fire and the `segments` property will have an array of 2 segments representing the two sides beside the fold/hinge. 

Just like the other properties of the visual viewport segments they represent the given state at the time `window.visualViewport` was queried.

Reply:
Right, I get that, the point is that my understanding is that `FrozenArray`s aren't supposed to change, ever, not even in reactions to an event. @domenic is that right?

Reply:
FrozenArray properties can change. The contents of the array cannot, but a new array can start being returned.

Reply:
Note that it is invalid for attributes to be sequences; that is part of why FrozenArray exists.